robot/robot.go

```diff
@@ -68,15 +72,35 @@ func (r *Robot) Move(steps int) error {
 	case West:
 		newPosition.X = r.Position.X - steps
 	}
-	if r.EnvMap.isObstacleAt(newPosition) {
-		return fmt.Errorf("The robot can't move %v steps because their are obstacles in the way.Here's the current map: %v", steps, mapWithRobot(r))
-	}
 	distance := Centimeters(float64(steps) * float64(r.EnvMap.SquareSize))
 	r.MoveModule.Move(distance, r.Tachometer)
 	r.Position = newPosition
 	return nil
 }
 
+func (r *Robot) canMoveToPosition(steps int) bool {
+	newPosition := r.Position
+	absSteps := int(math.Abs(float64(steps)))
+	singleStep := steps / absSteps
+	for i := 0; i < absSteps; i++ {
+		switch r.Position.Orientation {
```

Comment:
Please make the switch a (not exported) function to reduce the cyclomatic complexity of the canMoveToPosition(int) method.